Stripe/STPTelemetryClient.swift

```diff
@@ -0,0 +1,139 @@
+//
+//  STPTelemetryClient.swift
+//  Stripe
+//
+//  Created by Ben Guo on 4/18/17.
+//  Copyright © 2016 Stripe, Inc. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+final class STPTelemetryClient: NSObject {
+  @objc(sharedInstance) static var shared: STPTelemetryClient = STPTelemetryClient(sessionConfiguration: STPAPIClient.sharedUrlSessionConfiguration)
+
+  func addTelemetryFields(toParams params: inout [String: Any]) {
+    params["muid"] = muid
+  }
+
+  @objc func paramsByAddingTelemetryFields(toParams params: [String: Any]) -> [String: Any] {
+    var mutableParams = params
+    mutableParams["muid"] = muid
+    return mutableParams
+  }
+
+  func sendTelemetryData() {
+    guard STPTelemetryClient.shouldSendTelemetry() else {
+      return
+    }
+    let path = "ios-sdk-1"
+    let url = URL(string: "https://m.stripe.com")!.appendingPathComponent(path)
+    let request = NSMutableURLRequest(url: url)
+    request.httpMethod = "POST"
+    request.setValue("application/json", forHTTPHeaderField: "Content-Type")
+    let payload = self.payload
+    var data: Data?
+    do {
+      data = try JSONSerialization.data(
+        withJSONObject: payload, options: JSONSerialization.WritingOptions(rawValue: 0))
+    } catch {
+      return
+    }
+    request.httpBody = data
+    let task = urlSession.dataTask(with: request as URLRequest)
+    task.resume()
+  }
+
+  private let urlSession: URLSession
+
+  class func shouldSendTelemetry() -> Bool {
+    #if targetEnvironment(simulator)
+      return false
+    #else
+      return StripeAPI.advancedFraudSignalsEnabled && NSClassFromString("XCTest") == nil
+    #endif
+  }
+
+  init(sessionConfiguration config: URLSessionConfiguration) {
+    urlSession = URLSession(configuration: config)
+    super.init()
+  }
+
+  var muid : String {
+    get {
+      let muid = UIDevice.current.identifierForVendor?.uuidString
+      return muid ?? ""
+    }
+  }
+
+  var language = Locale.autoupdatingCurrent.identifier
```

Comment:
Are these properties supposed to be `private`? They aren't exposed in `STPTelemetryClient.h`

Reply:
They should be, but in the interest of time we haven't really been enforcing internal vs private boundaries. I'll fix this file up, at least.